cohttp-lwt/src/s.ml

```diff
@@ -13,6 +13,19 @@ module type IO = sig
       which case it returns the error. *)
 
   val pp_error : Format.formatter -> error -> unit
+
+  val wait_eof_or_closed : conn -> ic -> (unit -> unit t) -> unit t
+  (** [wait_eof_or_closed conn ic sleep_fn] waits for an EOF or a Closed status
+      on the input channel [ic]. This function is designed to be used in
+      [Lwt.pick] to run concurrently with the request handling from the input
+      channel. [sleep_fn] is a function used to yield control periodically,
+      keeping the Cohttp platform independent. This function ensures that the
+      monitoring does not spin too quickly and uses CPU efficiently when the
+      input channel has read activity but the client is not reading it. The
+      function checks for EOF using [MSG_PEEK] on the input channel without
+      consuming data, thereby not disturbing the request handling. If the
+      connection is closed locally, Cohttp will stop waiting for EOF and will
+      wait the promise to be cancelled.*)
```

Comment:
In this block of text, the subject switches back and forth between the `wait_eof_or_closed` function and the `sleep_fn` function. Confusingly, they are referred to with ambiguous demontratives (`wait_eof_or_closed`, This function, The function vs `sleep_fn`, This function).

I'd recommend writing first a paragraph focused only on the `wait_eof_or_closed` function, then a paragraph about the special case of the local close, and then finally a paragraph about the `sleep_fn` parameter (in which you use "This parameter" as a demonstrative rather than "This function" to keep things even clearer).

Reply:
Rephrased in [fixup1](https://github.com/gabrielmoise17/ocaml-cohttp/pull/3/commits/b9a2a286f87995beb19446b72c55760e89b5e5a9). Hope it is more informative/explicit now :)